### DISCUSSION TITLE
COAP observe command missing in COAP CLI

### DISCUSSION THREAD

**Discussion by bilalahmaddev** [openthread/openthread#9029] [Q&A] [answered]
I am not able to see COAP observe command in COAP CLI command list:
![image](https://user-images.githubusercontent.com/9848328/236631801-20f89f7d-7242-4cb7-bfb1-e82457f3edf2.png)
I am using this version of OpenThread:
![image](https://user-images.githubusercontent.com/9848328/236631840-6c46f909-22bd-427d-8646-c5314a0583ec.png)

**Comment by jwhui** [comment, marked as answer]
CoAP Observe is an optional build feature. See the `OT_COAP_OBSERVE` CMake option.

https://github.com/openthread/openthread/blob/eaa2779047c3bf781874954e82c716239460c5f7/etc/cmake/options.cmake#L96

**Reply by bilalahmaddev** [reply]
how to enable it?

**Reply by jwhui** [reply]
In general, enable CMake build options with `-DOT_COAP_OBSERVE=on`. Exactly where to place it depends on how you're building it (whether specifying everything in command line, using convenience build scripts, etc.).

**Comment by bilalahmaddev** [comment]
I am using ot-br-posix on raspberry pi

**Reply by jwhui** [reply]
You can modify this file directly: https://github.com/openthread/ot-br-posix/blob/main/third_party/openthread/CMakeLists.txt

Or you can prepend when calling the setup script:
```
OTBR_OPTIONS='-DOT_COAP_OBSERVE=on' ./script/setup
```

BTW, please respond to the previous thread rather than proposing a new answer.
